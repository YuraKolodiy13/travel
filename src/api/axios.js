import axios from "axios";

const axiosInstance = axios.create();

axiosInstance.interceptors.response.use(
  (response) => {

    return Promise.resolve(response);
  },
  (error) => {
    return Promise.reject(error);
  }
);

export {axiosInstance as axios};
