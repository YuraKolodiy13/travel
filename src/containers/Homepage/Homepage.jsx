import React, {memo, useEffect} from "react";
import {useDispatch, useSelector} from "react-redux";
import {SEARCH_FORM_REQUEST} from "../../actions/general";
import {GET_HOT_TOURS_REQUEST, GET_RECOMMENDED_TOURS_REQUEST} from "../../actions/homepage";
import Search from "../../components/Search/Search";
import HotelCard from "../../components/HotelCard/HotelCard";
// import Input from "../../components/Input/Input";
import Loader from "../../components/Loader/Loader";
import {DEFAULT_SEARCH_VALUE} from "../../helpers/constants";

const Homepage = () => {

  const hotTours = useSelector((state) => state.homepage.hotTours);
  const loading = useSelector((state) => state.general.loading);
  const recommendedTours = useSelector((state) => state.homepage.recommendedTours);
  const flights = useSelector((state) => state.general.flights);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch({
      type: SEARCH_FORM_REQUEST,
      payload: DEFAULT_SEARCH_VALUE
    })
    dispatch({
      type: GET_HOT_TOURS_REQUEST,
      payload: {
        body: {
          "landingId": null,
          "country": -1,
          "CountryAlpha": "vse-strany",
          "operatorIntId": null,
          "regions": [],
          "resorts": [],
          "hotels": [],
          "townFrom": "LWO",
          "cityFromCode": "LWO",
          "cityFrom": 5,
          "viewType": "horizontal",
          "checkIn": null,
          "dateFrom": null,
          "systemKey": null,
          "partnerType": null,
          "nightFrom": null,
          "nightTo": null
        }
      }
    })
    dispatch({
      type: GET_RECOMMENDED_TOURS_REQUEST,
      payload: {
        body: {
          "landingId": null,
          "country": -1,
          "CountryAlpha": "vse-strany",
          "operatorIntId": null,
          "regions": [],
          "resorts": [],
          "hotels": [],
          "townFrom": "LWO",
          "cityFromCode": "LWO",
          "cityFrom": 5,
          "viewType": "horizontal",
          "checkIn": null,
          "dateFrom": null,
          "systemKey": null,
          "partnerType": null,
          "nightFrom": null,
          "nightTo": null
        }
      }
    })
  }, [dispatch]);


  return (
    <div className='Homepage'>
      {!loading
        ? <>
            {/*<Input value='5' onChange={() => {}}/>*/}
            <Search />
            {!!hotTours.length && (
              <div className="hotTours">
                <h3>Гарячі тури</h3>
                {hotTours?.map(item =>
                  <HotelCard item={item} key={item.hotelId} flights={flights[item.SystemKey]}/>
                )}
              </div>
            )}
            {!!recommendedTours.length && (
              <div className="recommendedTours">
                <h3>Рукомендовані тури</h3>
                {recommendedTours?.map(item =>
                  <HotelCard item={item} key={item.hotelId} flights={flights[item.SystemKey]}/>
                )}
              </div>
            )}
          </>
        : <Loader/>
      }

    </div>
  )
};

export default memo(Homepage);