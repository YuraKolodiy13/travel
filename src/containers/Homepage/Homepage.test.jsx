import Homepage from "./Homepage";
import {render} from '@testing-library/react';
import React from "react";
import {MemoryRouter} from "react-router";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import axios from 'axios';

describe('testing homepage', () => {
  const mockStore = configureStore();
  let store = mockStore({general: {
      loading: false,
      searchForm: {},
      hash: '',
      hotels: null,
      tour: {},
      otherTours: [],
      hotTours: [ {
        disablePromoCode: false,
        HotType: 1,
        EarlyType: 1,
        RecommendedType: 1,
        isHot: false,
        isPromo: false,
        isBestDeal: false,
        LoadedDate: '2020-01-01T00:00:00+02:00',
        operatorName: 'JoinUp',
        operatorIdInt: 110,
        operatorId: '{JoinUpNativeApi}',
        type: 'plain',
        cityFromCode: 'KBP',
        cityFromName: 'Киев',
        flyInclude: 'true',
        countryID: '25',
        countryName: 'Египет',
        countryNameAccusativeCase: 'в Египет',
        checkIn: {
          value: '2022-09-13T00:00:00+03:00',
          hash: -1460628963
        },
        region: {
          RegionName: 'Шарм-эль-Шейх',
          ResortName: 'Шарм-эль-Шейх',
          RegionHash: 621,
          ResortHash: 1486,
          value: 'Шарм-эль-Шейх Шарм-эль-Шейх',
          hash: -2099355937
        },
        hotel: {
          hash: -641217413,
          value: 'Il Mercato Hotel (ex.Iberotel Il Mercato)'
        },
        hotelId: 34256,
        hotelUrl: '/hotel/eg/il-mercato-hotel',
        hotelKey: '34256',
        star: {
          hash: 7,
          value: '5'
        },
        meal: {
          hash: 1,
          value: 'Все включено (AI)'
        },
        room: 'Standard Room',
        htplace: 'DBL',
        price: 1561,
        customPriceWd: 0,
        customPriceWdUah: 0,
        currency: 'USD',
        priceUAH: 45660,
        nights: 14,
        address: 'Шарм-эль-Шейх',
        photo: 'https://img4.farvater.travel/mapkey/34256/0?size=catalog',
        rate: '8',
        adl: 2,
        kids: 0,
        ages: '',
        chd: 0,
        isEarly: false,
        SystemKey: '2m5136450731218604758c25',
        othersAgr: [],
        other: [],
        isRecommended: true,
        IsChoiceFarvater: false,
        FarTourId: 72265714,
        sort: 0,
        IsFreePCRTest: false,
        AdditionalInfo: null,
        idsForText: {
          roomId: 32,
          htplaceId: 334006,
          mealId: 1,
          starId: 7,
          countryid: 25,
          longhotelId: 34256,
          airportId: 4,
          transferTypeId: 1,
          touristsBits: 2,
          reviewsCount: '329',
          currencyId: 4
        },
        redPriceUAH: 45660,
        countOther: 1,
        latitude: 27.8633414,
        longitude: 34.3074194,
        next: null,
        prev: null,
        tourFreeCancel: false,
        isOtp: true,
        HealthCertificate: false,
        WorksIn2020: false,
        promotionEndDate: null,
        isLastSeats: false
      }],
      recommendedTours: [],
      flights: {}
    }})

  test('testing if requests was launched', () => {
    // axios.get.mockReturnValue(store);
    const {getByText} = render(
      <Provider store={store}>
        <MemoryRouter>
          <Homepage/>
        </MemoryRouter>
      </Provider>
      )
    expect(getByText('Гарячі тури')).toBeInTheDocument();
  })

  test('snapshot', () => {
    const homepage = render(
      <Provider store={store}>
        <MemoryRouter>
          <Homepage/>
        </MemoryRouter>
      </Provider>
      )
    expect(homepage).toMatchSnapshot()
  })

})