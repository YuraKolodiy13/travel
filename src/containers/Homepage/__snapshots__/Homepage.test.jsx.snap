// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing homepage snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Homepage"
      >
        <form
          class="search-form"
        >
          <div>
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="mui-5"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-5"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="cityFrom"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-nnbavb"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div>
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="mui-6"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-6"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="destination"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-nnbavb"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="css-17vdz66"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="mui-7"
                id="mui-7-label"
              >
                Start
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-7"
                  name="date"
                  placeholder="mm/dd/yyyy"
                  type="tel"
                  value="08/10/2022-08/24/2022"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1z7n62"
                  >
                    <span>
                      Start
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <button
            class="button primary"
            type="submit"
          >
            <span>
              Найти
            </span>
          </button>
        </form>
        <div
          class="hotTours"
        >
          <h3>
            Гарячі тури
          </h3>
          <div
            class="HotelCard"
          >
            <div
              class="hotelCard__img"
            >
              <img
                alt=""
                src="https://img4.farvater.travel/mapkey/34256/0?size=catalog"
              />
            </div>
            <div
              class="hotelCard__info"
            >
              <div
                class="hotelCard__price"
              >
                <a
                  href="/countries/eg/il-mercato-hotel?q=2m5136450731218604758c25"
                >
                  Il Mercato Hotel (ex.Iberotel Il Mercato)
                </a>
                <span>
                     $
                  1,561
                </span>
                /
                <span>
                  45,660
                  грн
                </span>
              </div>
              <ul>
                <li>
                  Включно з перельотом (
                  Киев
                  )
                </li>
                <li>
                  Виліт на відпочинок 
                  
                  12
                  .
                  09
                  <span
                    class="hotelCard__time"
                  >
                    Час
                  </span>
                </li>
                <li>
                  На 
                  14
                   ночей
                </li>
                <li>
                  Ціна за 
                  2
                   туриста
                </li>
                <li>
                  Standard Room
                </li>
                <li>
                  Все включено (AI)
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Homepage"
    >
      <form
        class="search-form"
      >
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-5"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-5"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="cityFrom"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-nnbavb"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-6"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-6"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="destination"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-nnbavb"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="css-17vdz66"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="mui-7"
              id="mui-7-label"
            >
              Start
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-7"
                name="date"
                placeholder="mm/dd/yyyy"
                type="tel"
                value="08/10/2022-08/24/2022"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1z7n62"
                >
                  <span>
                    Start
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          class="button primary"
          type="submit"
        >
          <span>
            Найти
          </span>
        </button>
      </form>
      <div
        class="hotTours"
      >
        <h3>
          Гарячі тури
        </h3>
        <div
          class="HotelCard"
        >
          <div
            class="hotelCard__img"
          >
            <img
              alt=""
              src="https://img4.farvater.travel/mapkey/34256/0?size=catalog"
            />
          </div>
          <div
            class="hotelCard__info"
          >
            <div
              class="hotelCard__price"
            >
              <a
                href="/countries/eg/il-mercato-hotel?q=2m5136450731218604758c25"
              >
                Il Mercato Hotel (ex.Iberotel Il Mercato)
              </a>
              <span>
                   $
                1,561
              </span>
              /
              <span>
                45,660
                грн
              </span>
            </div>
            <ul>
              <li>
                Включно з перельотом (
                Киев
                )
              </li>
              <li>
                Виліт на відпочинок 
                
                12
                .
                09
                <span
                  class="hotelCard__time"
                >
                  Час
                </span>
              </li>
              <li>
                На 
                14
                 ночей
              </li>
              <li>
                Ціна за 
                2
                 туриста
              </li>
              <li>
                Standard Room
              </li>
              <li>
                Все включено (AI)
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
